### PR TITLE
Corrected the way to compute distance from soma in latest NEURON version

### PR DESCRIPTION
**Pull request by pedroernesto**
Corrected the way to compute distance from soma in latest NEURON version, plus Python 3 compatibility

### CODE REVIEW

**Comment by robban80** [comment]
@tommorse could you please merge this pull request that fixes an outdated way of using the NEURON "distance" function?

File: fig2_validation.py [hunk 1]
```diff
@@ -46,212 +46,197 @@ def main(   par="./params_dMSN.json",        \
...
     # set pointer target in cascade
-    pointer =   casc._ref_Target1p    
-       
-    
-    # set edge of soma as reference for dendritic distance 
+    pointer =   casc._ref_Target1p
+
+
+    # set edge of soma as reference for dendritic distance
     h.distance(1, sec=h.soma[0])
```
Comment (ramcdougal) [inline fig2_validation.py:65]: is this line still relevant? Note that here we're measuring distance from the 1 edge of the soma, whereas the more modern use of `h.distance` is measuring from the center in `cell.soma(0.5)`. Are `h.soma[0]` and `cell.soma` the same section or different sections?

File: Results/Ca/ca_0_ca_soma0_0.out [hunk 2]
```diff
@@ -0,0 +1,8002 @@
+0 2e-05
```
Comment (ramcdougal) [inline Results/Ca/ca_0_ca_soma0_0.out]: Are all the `.out` files generated data? If so, is there any reason to include them in the source code?

Comment (ramcdougal) [reply Results/Ca/ca_0_ca_soma0_0.out]: Same question for the pdfs under `Figures`

File: fig2_validation.py [hunk 3]
```diff
@@ -28,15 +28,15 @@
 h.load_file('import3d.hoc')
 
 
- 
-    
+
```
Comment (ramcdougal) [inline fig2_validation.py]: Can you provide this change without whitespace changes? These edits hide the substantive changes... or if you want the whitespace edits, can you make that in a separate commit?

File: fig2_validation.py [hunk 4]
```diff
@@ -46,212 +46,197 @@ def main(   par="./params_dMSN.json",        \
...
     for sec in h.allsec():
-        
+
         for seg in sec:
-            
-            
+
+
             # naf and kas is in all sections
             h.setpointer(pointer, 'pka', seg.kas )
```
Comment (ramcdougal) [inline fig2_validation.py:93]: This below probably wouldn't have worked when the model was being developed (so I wouldn't change it here), but for the record with recent versions of NEURON, you could just say:
```python
seg.kas._ref_pka = pointer
```

**Comment by ramcdougal** [comment]
@robban80 is @pedroernesto part of your team? Can you please review his changes, the comments above, and advise? (Tom has moved on to other projects.)

File: Results/Ca/ca_0_ca_soma0_0.out (hunk 2 above)
Comment (pedroernesto) [reply Results/Ca/ca_0_ca_soma0_0.out]: For the execution of the NEURON code you will need the directories `Results/Ca` and `Results/FI` and `Figures` but not the figures.

**Comment by pedroernesto** [comment]
> I was part of the HBP model validation team. To make the model run in modern NEURON, I needed to introduce the changes commented.

File: Results/Ca/ca_0_ca_soma0_0.out (hunk 2 above)
Comment (robban80) [reply Results/Ca/ca_0_ca_soma0_0.out]: Possibly we could add the directories - if they are needed and missing - but skip the .out and .pdf files?

File: fig2_validation.py (hunk 4 above)
Comment (robban80) [reply fig2_validation.py:93]: thanks for the advice!

File: Results/Ca/ca_0_ca_soma0_0.out (hunk 2 above)
Comment (ramcdougal) [reply Results/Ca/ca_0_ca_soma0_0.out]: If all you need is the directories, one way to do that is to do something like:

```python
import os
for name in ["Results/Ca", "Results/FI", "Figures"]:
    try:
        os.makedirs(name)
    except:
        pass
```

**Comment by robban80** [comment]
I have reviewed the changes. Perhaps the generated data and pdf files could be excluded from the commits, but apart from that I think they are fine. I can also confirm that Pedro has helped to update the model so that it is runnable with never version of NEURON.

Thanks for the thorough code review @ramcdougal !

File: fig2_validation.py (hunk 3 above)
Comment (pedroernesto) [reply fig2_validation.py]: Done, preparing new commit

File: Results/Ca/ca_0_ca_soma0_0.out (hunk 2 above)
Comment (pedroernesto) [reply Results/Ca/ca_0_ca_soma0_0.out]: Ok, added this in file fig2_validation.py

**Comment by pedroernesto** [comment]
Changes suggested in new commit.

**Comment by ramcdougal** [comment]
Are you making a new pull request?

**Comment by pedroernesto** [comment]
Yes, I made another pull request
